compaction/compaction_manager.cc

```diff
@@ -1240,13 +1240,20 @@ class compaction_manager::validate_sstables_compaction_task : public compaction_
     }
 };
 
-future<> compaction_manager::perform_sstable_scrub_validate_mode(replica::table* t) {
+static std::vector<sstables::shared_sstable> get_all_sstables(compaction::table_state& t) {
+    auto s = boost::copy_range<std::vector<sstables::shared_sstable>>(*t.main_sstable_set().all());
```

Comment:
I don't think you  need `boost::copy_range()` here. `std::vector` has a copy constructor, including one which operates on a pair of iterators.

Reply:
I see you copied this from below. I think this possibly had a filtering sometimes in the past, the filtering was removed and the `boost::copy_range()` survived.

Reply:
You're spot on :-) will remove it. thanks

Reply:
actually, turns out it's because sstable_set::all() 's return type is unordered_set.